From 829e2325384b0f8b8e299ddc3e572281cdb5ca2f Mon Sep 17 00:00:00 2001
From: John Cox <jc@kynesim.co.uk>
Date: Tue, 18 Oct 2022 14:07:04 +0000
Subject: [PATCH 154/222] v4l2_m2m: Add function to get number of queued
 buffers

(cherry picked from commit f9ac6485c00b4531dcff354222aef450b29728f4)
(cherry picked from commit ce03804363532605e64cccdc70ce0c71739a8415)
---
 libavcodec/v4l2_context.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/libavcodec/v4l2_context.h b/libavcodec/v4l2_context.h
index 311b6f10a4..613c3f3d06 100644
--- a/libavcodec/v4l2_context.h
+++ b/libavcodec/v4l2_context.h
@@ -218,4 +218,15 @@ int ff_v4l2_context_enqueue_frame(V4L2Context* ctx, const AVFrame* f);
 
 void ff_v4l2_dq_all(V4L2Context *const ctx);
 
+/**
+ * Returns the number of buffers currently queued
+ *
+ * @param[in] ctx The V4L2Context to evaluate
+ */
+static inline int
+ff_v4l2_context_q_count(const V4L2Context* const ctx)
+{
+    return atomic_load(&ctx->q_count);
+}
+
 #endif // AVCODEC_V4L2_CONTEXT_H
-- 
2.45.2

